compiler/aarch64/runtime/ARM64arrayCopy.spp

```diff
@@ -92,14 +158,17 @@ fwDstAlign8:
 fwDstAlign16:
 	tst	x1, #8
 	bne	fwDoubleWordCopyLoop		// srcAddr is not 16-byte aligned
-fwQuadWordCopyLoop:
+__fwQuadWordCopyLoop:
+	cbz	x0, finished			// return if no bytes to copy
+	subs	x3, x2, x1
+	beq	finished			// return if srcAddr == dstAddr
 	// Both srcAddr and dstAddr are 16-byte aligned
 	cmp	x0, #16
 	blt	fwDoubleWordCopy		// less than 16 bytes remaining
 	ldp	x3, x4, [x1], #16
 	sub	x0, x0, #16
 	stp	x3, x4, [x2], #16
-	b	fwQuadWordCopyLoop
+	b	__fwQuadWordCopyLoop
```

Comment:
You perform redundant checks for `x0 != 0` and `x1 != x2` every time in the loop by jumping to `__fwQuadWordCopyLoop`.
Have you checked the arraycopy performance with and without your changes?

Do it in the same way as other entry points such as `__fwDoubleWordArrayCopy`.

Reply:
The updated code is still wrong.

- Original code: 6 instructions for copying 16 bytes of array data in `fwQuadWordCopyLoop`.
- Updated code: 9 instructions for doing the same in `__fwQuadWordCopyLoop`.